components/org.wso2.carbon.identity.application.authenticator.samlsso/src/main/java/org/wso2/carbon/identity/application/authenticator/samlsso/logout/processor/SAMLLogoutRequestProcessor.java

```diff
@@ -0,0 +1,243 @@
+
+        AuthenticationRequest authenticationRequest = new AuthenticationRequest();
+        authenticationRequest.appendRequestQueryParams(parameterMap);
+        if (identityRequest.getHeaderMap() != null) {
+            identityRequest.getHeaderMap().forEach(authenticationRequest::addHeader);
+        }
+        authenticationRequest.setTenantDomain(samlMessageContext.getTenantDomain());
+        authenticationRequest.setRelyingParty(getRelyingPartyId(samlMessageContext));
+        authenticationRequest.setType(getType(samlMessageContext));
+        try {
+            authenticationRequest.setCommonAuthCallerPath(URLEncoder.encode(getCallbackPath(samlMessageContext),
+                    StandardCharsets.UTF_8.name()));
+        } catch (UnsupportedEncodingException e) {
+            throw FrameworkRuntimeException.error("Error occurred while URL encoding callback path: " +
+                    getCallbackPath(samlMessageContext), e);
+        }
+        authenticationRequest.addRequestQueryParam(LOGOUT, new String[]{TRUE});
+        authenticationRequest.addRequestQueryParam(SESSION_ID, new String[]{samlMessageContext.getSessionID()});
+
+        AuthenticationRequestCacheEntry authRequest = new AuthenticationRequestCacheEntry(authenticationRequest);
+        String sessionDataKey = UUIDGenerator.generateUUID();
+        authRequest.setValidityPeriod(TimeUnit.MINUTES.toNanos(IdentityUtil.getOperationCleanUpTimeout()));
+        FrameworkUtils.addAuthenticationRequestToCache(sessionDataKey, authRequest);
+
+        InboundUtil.addContextToCache(sessionDataKey, samlMessageContext);
+
+        FrameworkLogoutResponse.FrameworkLogoutResponseBuilder responseBuilder =
+                new FrameworkLogoutResponse.FrameworkLogoutResponseBuilder(samlMessageContext);
+        responseBuilder.setContextKey(sessionDataKey);
+        responseBuilder.setCallbackPath(getCallbackPath(samlMessageContext));
+        responseBuilder.setAuthType(getType(samlMessageContext));
+        String commonAuthURL = IdentityUtil.getServerURL(COMMONAUTH, true, true);
+        responseBuilder.setRedirectURL(commonAuthURL);
+        return responseBuilder;
+    }
+
+    /**
+     * Populate SAMLMessageContext with session details of the SAML index.
+     *
+     * @param samlMessageContext {@link SAMLMessageContext} object which has details on logout flow.
+     * @throws SAMLIdentityException Error while retrieving the session details.
+     */
+    private void populateContextWithSessionDetails(SAMLMessageContext<String, String> samlMessageContext)
+            throws SAMLIdentityException {
+
+        SessionInfoDAO sessionInfoDAO = new SessionInfoDAO();
+        Map<String, String> sessionDetails = sessionInfoDAO.getSessionDetails
+                (samlMessageContext.getIdPSessionID());
+        if (sessionDetails != null) {
+            if (StringUtils.isNotBlank(samlMessageContext.getSAMLLogoutRequest().getTenantDomain())) {
```

Comment:
better to assign `samlMessageContext.getSAMLLogoutRequest().getTenantDomain()` to a variable and resuse

Reply:
Fixed with PR https://github.com/wso2-extensions/identity-outbound-auth-samlsso/pull/133